x/attribute/keeper/msg_server.go

```diff
@@ -203,3 +222,42 @@ func (k msgServer) DeleteDistinctAttribute(goCtx context.Context, msg *types.Msg
 
 	return &types.MsgDeleteDistinctAttributeResponse{}, nil
 }
+
+// setExpirationForNameRecord registers an expiration record
+// with the expiration module for the supplied types.NameRecord
+func (k msgServer) setExpirationForAttribute(
+	ctx sdk.Context,
+	attr types.Attribute,
+	owner sdk.AccAddress,
+	expPeriod string,
+) error {
+	// create expire name request
+	expireMsg := types.NewMsgDeleteAttributeRequest(attr.Address, owner, attr.Name)
+	wrapper, anyErr := codectypes.NewAnyWithValue(expireMsg)
+	if anyErr != nil {
+		return anyErr
+	}
+
+	// if an expiration period isn't provided, use default expiration period
+	expShort := expPeriod
+	if len(expShort) == 0 {
+		expShort = k.Keeper.GetDefaultAttributeExpiration(ctx, attr)
+	}
+
+	// parse expiration into a duration
+	duration, durErr := exptypes.ParseDuration(expShort)
+	if durErr != nil {
+		return durErr
+	}
+
+	// create expiration metadata that will expire scope when executed.
+	expTime := ctx.BlockTime().Add(*duration)
+	expDeposit := sdk.NewCoins(k.expKeeper.GetDeposit(ctx))
+	expiration := exptypes.NewExpiration(owner.String(), owner.String(), expTime, expDeposit, *wrapper)
```

Comment:
The owner address is not a good thing to use as the `ModuleAssetID`. Doing so means they can only ever have one expiration. If someone owns multiple attributes, this falls apart.